### PR TITLE
3 functions added to user.py

### PR DESCRIPTION
**Pull request by glebmiller**
This functions allow to get:
1) dict of all tittles the user have with completage percentage
2) id's of earned trophies of specific title
3) names and types of trophies for specific title

### CODE REVIEW

File: src/psnawp_api/models/user.py [hunk 1]
```diff
@@ -250,6 +250,75 @@ def is_blocked(self) -> bool:
         ).json()
         return self.account_id in response["blockList"]
 
+    def get_all_trophies(self, limit=799):
```
Comment (Lakrix) [inline src/psnawp_api/models/user.py:253]: I think the max throphies returned are 100. You can probably pass an offset aswell to support a way to get all thropies.

File: src/psnawp_api/models/user.py [hunk 2]
```diff
@@ -250,6 +250,75 @@ def is_blocked(self) -> bool:
...
+            limit (int, optional): limit of trophies to query. Max is 800. Defaults to 100.
+
+        Returns:
+            dict
+        """
+
+        param = {"limit": min(limit, 800)}
+
+        response = self._request_builder.get(
+            url=f"{User.trophy_base_uri}/v1/users/{self.account_id}/trophyTitles",
```
Comment (toymak3r) [inline src/psnawp_api/models/user.py:266]: This "user.trophy_base_uri" should be replaced by BASE_PATH['trophy_base_uri'] and BASE_PATH should include the right uri path.

File: src/psnawp_api/models/user.py (hunk 1 above)
Comment (glebmiller) [reply src/psnawp_api/models/user.py:253]: i've tested it and believe the number of returned titles is much more. had success with profiles with 400+ titles

File: src/psnawp_api/models/user.py (hunk 2 above)
Comment (glebmiller) [reply src/psnawp_api/models/user.py]: fixed that in new commit

**Comment by isFakeAccount** [comment]
I am closing this PR since the way I have implemented it is quite different. Since it is a feature requested a lot, I wanted to make it similar to how it behaves on other API wrappers. 

Check out the branch [trophy_support](https://github.com/isFakeAccount/psnawp/tree/trophy_support)

If you still want to contribute, check out the https://github.com/isFakeAccount/psnawp/blob/trophy_support/src/psnawp_api/models/title.py

The endpoints are already there in the `endpoint.py` file. You can check `trophy_titles.py` to see how the pagination support is implemented. 

I will try to work on it this weekend anyways. Hopefully, I can get the test cases done too.

**Pull request closed without merging**